zephyr/CMakeLists.txt

```diff
@@ -115,7 +127,38 @@ if(CONFIG_WOLFSSL)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/psa/psa_hash.c)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/psa/psa_pkcbs.c)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/st/stm32.c)
-  
+
+    if(CONFIG_WOLFCRYPT_ARMASM)
+      # tested with board: "qemu_kvm_arm64"
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-aes.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-32-aes-asm_c.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-sha256.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-32-sha256-asm_c.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-sha512.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-sha512-asm_c.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-sha3-asm_c.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-poly1305.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-chacha.c)
+      zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/port/arm/armv8-curve25519_c.c)
+
+      set(MCPU_FLAGS "-mcpu=cortex-a53+crypto -mstrict-align")
```

Comment:
Not sure by this is not overriding the option correctly. I am getting error from assember :
/tmp/cclwNdTX.s: Assembler messages:
/tmp/cclwNdTX.s:58: Error: selected processor does not support `sha256h q12,q13,v0.4s'

Reply:
Updated my Zephyr cmake folder setting to run crypto instruction on my A53 core : 

diff --git a/cmake/gcc-m-cpu.cmake b/cmake/gcc-m-cpu.cmake
index ae10132dfc0..3f54c7384b6 100644
--- a/cmake/gcc-m-cpu.cmake
+++ b/cmake/gcc-m-cpu.cmake
@@ -72,7 +72,7 @@ if("${ARCH}" STREQUAL "arm")
   endif()
 elseif("${ARCH}" STREQUAL "arm64")
   if(CONFIG_CPU_CORTEX_A53)
-    set(GCC_M_CPU cortex-a53)
+    set(GCC_M_CPU cortex-a53+crypto -mstrict-align)
   elseif(CONFIG_CPU_CORTEX_A55)
     set(GCC_M_CPU cortex-a55)
   elseif(CONFIG_CPU_CORTEX_A76)